provider/ansible.py

```diff
@@ -170,7 +170,13 @@ def distro_install(packages="ansible"):
     package_list = params.get_list("package_list", 'sshpass')
     try:
         path.find_command('ansible-playbook')
-    except path.CmdNotFoundError:
+        # If ansible debug plugin is required we need to enforce full
+        # ansible version installation instead of ansible-core
+        if params.get("ansible_callback_plugin") == "debug":
```

Comment:
Leave this patch,  IMO, the current usage for `ansible_callback_plugin` is not very reasonable, it only be used in `ANSIBLE_STDOUT_CALLBACK`, but from the meaning of the param name `ansible_callback_plugin`, not stdout only：
https://github.com/autotest/tp-qemu/blob/32cb1c78f39342f1793db534107ae1effbeb7efc/provider/ansible.py#L63

How do you think about using `if 'debug' in params.objects('ansible_callback_plugin'):`

Reply:
In the past ansible versions, the output of `ansible-doc -t callback -l` does not have a prefix, just like `default`, `debug`. So to make the check more common, I would like to update it as:
```
cmd = 'ansible-doc -t callback -l | grep -qE "^(ansible.[a-z]+\.?)?%s"' % ansible_callback_plugin
```

I cannot ensure the pattern is correct, but it works for me.
```bash
$ echo "ansible.builtin.default default Ansible screen output" | grep -E "^(ansible.[a-z]+\.?)?default"
ansible.builtin.default default Ansible screen output
$ echo "ansible.builtin.default default Ansible screen output" | grep -E "^(ansible.[a-z]+\.?)?debug"

$ echo "default default Ansible screen output" | grep -E "^(ansible.[a-z]+\.?)?default"
default default Ansible screen output
$ echo "ansible.builtin.default default Ansible screen output" | grep -E "^(ansible.[a-z]+\.?)?ansible.builtin.default"
ansible.builtin.default default Ansible screen output
$ echo "ansiblea.builtin.default default Ansible screen output" | grep -qE "^(ansible.[a-z]+\.?)?default"
```

Reply:
I have no problem with that, let me change that :+1:

Reply:
@PaulYuuu Thanks for the catch. I didn't consider that. Regarding the pattern, I think it is fine :+1:  and it also works correctly for me. (I just included the change.)